### ISSUE TITLE
No executable is listening on host port 

### ISSUE THREAD

**Issue by pendenaor** [docker/for-win#10424]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 4AD08EC8-83BF-424D-B01E-64AC923AE8F6/20210219091544

### Actual behavior
Browser can't navigate to https://localhost:49157, "Connection failed"

### Expected behavior
Browser can navigate to https://localhost:49157

### Information

  - Windows Version: 10 Pro 20H2
  - Docker Desktop Version: 3.1.0
  - WSL2 or Hyper-V backend?  WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

I'm developpoing a WebApi with vs2019 without trouble for a long time but problems arise few days ago.
VS2019 build image, launch container and attach debugger without issue.

Container is up and webapi is listening to 49157 --> https
```
>docker container ls
CONTAINER ID   IMAGE                    COMMAND               CREATED          STATUS          PORTS
               NAMES
1897c75afffa   copperfieldwebhost:dev   "tail -f /dev/null"   11 minutes ago   Up 11 minutes   0.0.0.0:49158->80/tcp, 0.0.0.0:49157->443/tcp   CopperField.WebHost
```

webapi is accessible from the inside 
```
>docker exec -it 1897c75afffa netstat -aon
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       Timer
tcp6       0      0 :::80                   :::*                    LISTEN      off (0.00/0/0)
tcp6       0      0 :::443                  :::*                    LISTEN      off (0.00/0/0)
Active UNIX domain sockets (servers and established)
Proto RefCnt Flags       Type       State         I-Node   Path
unix  2      [ ACC ]     STREAM     LISTENING     362866   /tmp/dotnet-diagnostic-569-2971126-socket
unix  2      [ ACC ]     STREAM     LISTENING     367679   /tmp/dotnet-diagnostic-582-2971280-socket
unix  2      [ ]         DGRAM                    361031

>docker exec -it 1897c75afffa curl -k -v https://localhost:443
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=localhost
*  start date: Nov 17 08:18:15 2020 GMT
*  expire date: Nov 17 08:18:15 2021 GMT
*  issuer: CN=localhost
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x562d3b27af90)
> GET / HTTP/2
> Host: localhost
> User-Agent: curl/7.64.0
> Accept: */*
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 100)!
< HTTP/2 404
< date: Fri, 19 Feb 2021 09:13:16 GMT
< server: Kestrel
< content-length: 0
<
* Connection #0 to host localhost left intact
```

but not from the outside of the container
```
>curl -k -v https://localhost:49157
* Rebuilt URL to: https://localhost:49157/
*   Trying ::1...
* TCP_NODELAY set
*   Trying 127.0.0.1...
* TCP_NODELAY set
* connect to ::1 port 49157 failed: Connection refused
* connect to 127.0.0.1 port 49157 failed: Connection refused
* Failed to connect to localhost port 49157: Connection refused
* Closing connection 0
curl: (7) Failed to connect to localhost port 49157: Connection refused

>netstat -aon | find /c /i "49157"
0
```  

### Steps to reproduce the behavior
Don't know how to reproduce it, sorry.

Dockerfile
```
#See https://aka.ms/containerfastmode to understand how Visual Studio uses this Dockerfile to build your images for faster debugging.

FROM mcr.microsoft.com/dotnet/aspnet:5.0-buster-slim AS base
WORKDIR /app
EXPOSE 80
EXPOSE 443

FROM mcr.microsoft.com/dotnet/sdk:5.0-buster-slim AS build
WORKDIR /src
COPY ["src/CopperField.WebHost/CopperField.WebHost.csproj", "src/CopperField.WebHost/"]
RUN dotnet restore "src/CopperField.WebHost/CopperField.WebHost.csproj"
COPY . .
WORKDIR "/src/CopperField.WebHost"
RUN dotnet build "CopperField.WebHost.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "CopperField.WebHost.csproj" -c Release -o /app/publish

FROM base AS final
WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "CopperField.WebHost.dll"]
````

**Comment by stephen-turner** [comment]
I think this might be the same as docker/for-mac#5247. Does it work if you only forward a single port?

**Comment by pendenaor** [comment]
Modified my project settings to bind only port 443
```
>docker container ls
CONTAINER ID   IMAGE                    COMMAND               CREATED          STATUS          PORTS                    NAMES
2e592e9425ce   copperfieldwebhost:dev   "tail -f /dev/null"   44 seconds ago   Up 43 seconds   0.0.0.0:49163->443/tcp   CopperField.WebHost
```
but it's doesn't change anything.

**Comment by pendenaor** [comment]
As a side note, no process is listening on port 49163.

**Comment by pendenaor** [comment]
today, running the container fails with `5>  docker: Error response from daemon: Ports are not available: listen tcp 0.0.0.0:65260: bind: An attempt was made to access a socket in a way forbidden by its access permissions.`

**Comment by pendenaor** [comment]
I reboot my PC, and everything is back to normal, debugging aspnet core in Docker is now working.
I think it may be related to suspend mode.
